cannon/mipsevm/multithreaded/mips.go

```diff
@@ -311,8 +311,8 @@ func (m *InstrumentedState) doMipsStep() error {
 	}
 	m.state.StepsSinceLastContextSwitch += 1
 
-	//instruction fetch
-	insn, opcode, fun := exec.GetInstructionDetails(m.state.GetPC(), m.state.Memory)
+	insn_detail := m.cached_decode[m.state.GetPC()/4]
```

Comment:
The cache fits 512M instructions, which should always suffice. But in case it doesn't, let's fallback to reading and decoding the instruction from memory if the PC is larger than the cache.